dmcontent/questions.py

```diff
@@ -26,15 +28,20 @@ def __init__(self, data, number=None, _context=None):
     def summary(self, service_data, inplace_allowed: bool = False) -> "QuestionSummary":
         return QuestionSummary(self, service_data)
 
-    def filter(self, context, dynamic=True, inplace_allowed: bool = False) -> Optional["Question"]:
+    def filter(self: T, context, dynamic=True, inplace_allowed: bool = False) -> Optional[T]:
         if not self._should_be_shown(context):
             return None
 
         if inplace_allowed:
             self._context = context
             return self
         else:
-            return ContentQuestion(self._data, number=self.number, _context=context)
+            content_question = ContentQuestion(self._data, number=self.number, _context=context)
```

Comment:
Instead of using the factory here could we just use `self.__class__` as a constructor?

Reply:
Good point. And we also found another place where `ContentQuestion` was unnecessary